### PR TITLE
Changed URL to `ncview` source code.

### PR DESCRIPTION
**Pull request by dipietrantonio**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ncview/package.py [hunk 1]
```diff
@@ -10,7 +10,7 @@ class Ncview(AutotoolsPackage):
     """Simple viewer for NetCDF files."""
 
     homepage = "https://cirrus.ucsd.edu/ncview/"
-    url = "ftp://cirrus.ucsd.edu/pub/ncview/ncview-2.1.7.tar.gz"
+    url = "https://cirrus.ucsd.edu/~pierce/ncview/ncview-2.1.9.tar.gz"
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/ncview/package.py]: Can we retrieve previous versions from this `url`?

**Review by alalazo — CHANGES_REQUESTED** [review]
```console
$ wget https://cirrus.ucsd.edu/~pierce/ncview/ncview-2.1.9.tar.gz
--2023-08-08 09:10:19--  https://cirrus.ucsd.edu/~pierce/ncview/ncview-2.1.9.tar.gz
Resolving cirrus.ucsd.edu (cirrus.ucsd.edu)... 169.228.227.180
Connecting to cirrus.ucsd.edu (cirrus.ucsd.edu)|169.228.227.180|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1264439 (1,2M) [application/x-gzip]
Saving to: ‘ncview-2.1.9.tar.gz’

ncview-2.1.9.tar.gz                                                  100%[===================================================================================================================================================================>]   1,21M  1,04MB/s    in 1,2s    

2023-08-08 09:10:21 (1,04 MB/s) - ‘ncview-2.1.9.tar.gz’ saved [1264439/1264439]

$ wget https://cirrus.ucsd.edu/~pierce/ncview/ncview-2.1.7.tar.gz
--2023-08-08 09:10:37--  https://cirrus.ucsd.edu/~pierce/ncview/ncview-2.1.7.tar.gz
Resolving cirrus.ucsd.edu (cirrus.ucsd.edu)... 169.228.227.180
Connecting to cirrus.ucsd.edu (cirrus.ucsd.edu)|169.228.227.180|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2023-08-08 09:10:38 ERROR 404: Not Found.

$ wget ftp://cirrus.ucsd.edu/pub/ncview/ncview-2.1.7.tar.gz
--2023-08-08 09:11:48--  ftp://cirrus.ucsd.edu/pub/ncview/ncview-2.1.7.tar.gz
           => ‘ncview-2.1.7.tar.gz’
Resolving cirrus.ucsd.edu (cirrus.ucsd.edu)... 169.228.227.180
Connecting to cirrus.ucsd.edu (cirrus.ucsd.edu)|169.228.227.180|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/ncview ... done.
==> SIZE ncview-2.1.7.tar.gz ... 1280624
==> PASV ... done.    ==> RETR ncview-2.1.7.tar.gz ... done.
Length: 1280624 (1,2M) (unauthoritative)

ncview-2.1.7.tar.gz                                                  100%[===================================================================================================================================================================>]   1,22M  1,12MB/s    in 1,1s    

2023-08-08 09:11:52 (1,12 MB/s) - ‘ncview-2.1.7.tar.gz’ saved [1280624]
```

File: var/spack/repos/builtin/packages/ncview/package.py (hunk 1 above)
Comment (dipietrantonio) [reply var/spack/repos/builtin/packages/ncview/package.py]: Hi @alalazo , in my last commit, I propose choosing the URL based on version. I tested it and it works!

Comment (alalazo) [reply var/spack/repos/builtin/packages/ncview/package.py]: Given that:
```
$ wget wget ftp://cirrus.ucsd.edu/pub/ncview/ncview-2.1.9.tar.gz
--2023-08-10 08:33:50--  http://wget/
Resolving wget (wget)... 127.0.0.1
Connecting to wget (wget)|127.0.0.1|:80... failed: Connection refused.
--2023-08-10 08:33:50--  ftp://cirrus.ucsd.edu/pub/ncview/ncview-2.1.9.tar.gz
           => ‘ncview-2.1.9.tar.gz’
Resolving cirrus.ucsd.edu (cirrus.ucsd.edu)... 169.228.227.180
Connecting to cirrus.ucsd.edu (cirrus.ucsd.edu)|169.228.227.180|:21... connected.
Logging in as anonymous ... Logged in!
==> SYST ... done.    ==> PWD ... done.
==> TYPE I ... done.  ==> CWD (1) /pub/ncview ... done.
==> SIZE ncview-2.1.9.tar.gz ... done.

==> PASV ... done.    ==> RETR ncview-2.1.9.tar.gz ... 
No such file ‘ncview-2.1.9.tar.gz’.
```
What is the motivation for this change?

**Review by alalazo — APPROVED** [review]

**Pull request merged**